src/package-manager.ts

```diff
@@ -78,7 +85,12 @@ export async function detectPackageManager(
   const detected = await findup(
     resolve(cwd || "."),
     async (path) => {
-      // 1. Use `packageManager` field from package.json
+      // 1. Use `packageManager` field from package.json / deno.json
+      const denoJSONPath = join(path, "deno.json");
+      if (existsSync(denoJSONPath)) {
```

Comment:
Since deno is not most common runtime today, perf-wise it would be nicer to move this check later IMO